src/mlpack/methods/ann/loss_functions/mean_absolute_error_impl.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file methods/ann/loss_functions/mean_absolute_error_impl.hpp
+ * @author Aakash Kaushik
+ *
+ * Implementation of the Mean Absolute Error function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_MEAN_ABSOLUTE_ERROR_IMPL_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_MEAN_ABSOLUTE_ERROR_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "mean_absolute_error.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+MeanAbsoluteError<InputDataType, OutputDataType>::MeanAbsoluteError(const bool reduction):
+  reduction(reduction)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+typename InputType::elem_type
+MeanAbsoluteError<InputDataType, OutputDataType>::Forward(
+    const InputType& input,
+    const TargetType& target)
+{
+  if (reduction)
+    return arma::accu(arma::abs((input - target) / input));
+    
+  return arma::accu(arma::mean(arma::abs((input - target) / input)));
```

Comment:
Instead of calculating the same thing, do it once.
See how the ```lossSum``` variable is used in that PR.
Restructure this as
```
if (reduction)
   return lossSum

return lossSum / xyz
```